src/main/scala/esmeta/analyzer/AbsTransfer.scala

```diff
@@ -428,11 +428,18 @@ case class AbsTransfer(sem: AbsSemantics) {
           st <- get
         } yield st.contains(l, v, field)
       case ESubstring(expr, from, to) =>
-        for {
-          v <- transfer(expr)
-          f <- transfer(from)
-          t <- transfer(to)
-        } yield v.substring(f, t)
+        to match {
```

Comment:
Try to use the `fold` function of the `Option[_]` type. Do not use curly braces if possible.